Core/src/org/sleuthkit/autopsy/casemodule/AddImageTask.java

```diff
@@ -233,6 +241,48 @@ private void commitOrRevertAddImageProcess(Case currentCase, List<String> errorM
                             ImageWriterService.createImageWriter(imageId, imageWriterSettings);
                         }
                         newDataSources.add(newImage);
+                        if (!StringUtils.isBlank(md5)) {
+                            try {
+                                newImage.setMD5(md5);
+                            } catch (TskCoreException | TskDataException ex) {
+                                /*
+                                 * Treat both exceptions as the same since this
+                                 * is a new image and the hash should not
+                                 * already be set.
```

Comment:
Is that true? I thought the MD5 was set by the ewf code in TSK when opening the image.

Reply:
Good catch! I was not thinking about the possibility that a user would enter hashes in the panel for an EWF data source. I will fix this.